app/assets/javascripts/controllers/cloud_volume/cloud_volume_form_controller.js

```diff
@@ -240,5 +241,12 @@ ManageIQ.angular.app.controller('cloudVolumeFormController', ['miqService', 'API
     miqService.sparkleOff();
   };
 
+  var idFromHref = function(href) {
+    // Explicitly compare to undefined because 0 could potentially be a valid ID.
+    if(href === undefined) return 'undefined';
+    // Parse ID from href, e.g. http://172.16.117.189:3000/api/providers/123000000000001.
+    return href.toString().replace(/\/$/, '').split('/').slice(-1)[0];
+  };
+
```

Comment:
@himdel I added some more checks to prevent crash on strange outputs like `undefined` or integer value. It's better if we don't crash and return invalid value because we're just filling the drop-down after all...

Reply:
I've just tested cloud volume creation (the only operation where user can actually modify drop-down value for manager) and it passes on correct value with valid payload like:

```
name=miha-volume&aws_encryption=false&incremental=false&force=false&storage_manager_id=123000000000003&emstype=...
```

In other words, it seems to work.

Reply:
Yeah, looks about right :)

slice with negative indexes seems to be supported everywhere, not seeing any issues :+1: